scheduler/src/cook/kubernetes/compute_cluster.clj

```diff
@@ -370,125 +369,134 @@
     (async/chan 1))
 
   (pending-offers [this pool-name]
-    (let [node-name->node (get @pool->node-name->node pool-name)]
-      (if-not (or (cc/autoscaling? this pool-name) node-name->node)
+    (let [state @state-atom]
+      (if-not (= state :running)
```

Comment:
Should add similar gating to kill-tasks above, only kill if running or draining.

Reply:
No. You shouldn't get to a state when you have a task you want to kill and a cluster is "deleted". But if you do, it seems good to be able to make things better and delete the task. So it seems useful in a weird situation and doesn't hurt otherwise.

Reply:
Sure, in theory, this situation shouldn't occur. However, you're possibly sending an API request to a cluster that's deleted, whose IP may be re-used by anything. That seems wrong.

Reply:
updated